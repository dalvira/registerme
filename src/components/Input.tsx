import React, {useState} from 'react';
import {TextInput, StyleSheet, Text, View} from 'react-native';
import {colors} from '../theme/colors';

interface Props {
  testID?: string;
  placeholder?: string;
  hint?: string;
  value?: string;
  onChange?: (val: string) => void;
  secureTextEntry?: boolean;
  autoCapitalize?: 'none' | 'sentences' | 'words' | 'characters';
  keyboardType?:
    | 'default'
    | 'number-pad'
    | 'decimal-pad'
    | 'numeric'
    | 'email-address'
    | 'phone-pad';
}

const Input = (props: Props) => {
  const [text, setText] = useState(props.value);

  const onChangeText = (val: string) => {
    props.onChange && props.onChange(val);
    setText(val);
  };

  return (
    <View style={styles.container}>
      <TextInput
        testID={props.testID}
        style={styles.input}
        placeholder={props.placeholder}
        value={text}
        onChangeText={onChangeText}
        autoCapitalize={props.autoCapitalize}
        secureTextEntry={props.secureTextEntry}
        keyboardType={props.keyboardType}
      />
      {props.hint ? <Text style={styles.hint}>*{props.hint}</Text> : null}
    </View>
  );
};

export default Input;

const styles = StyleSheet.create({
  container: {
    marginBottom: 20,
  },
  input: {
    height: 40,
    width: 320,
    backgroundColor: colors.white,
    borderRadius: 8,
    paddingLeft: 10,
  },
  hint: {
    fontSize: 11,
    color: colors.darkGray,
    marginTop: 5,
    marginLeft: 5,
  },
});
