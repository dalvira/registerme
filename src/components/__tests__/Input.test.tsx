import React from 'react';
import renderer from 'react-test-renderer';
import Input from '../Input';

const mockCall = jest.fn();

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Input placeholder="Email" value="test@test.com" onChange={mockCall} />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
