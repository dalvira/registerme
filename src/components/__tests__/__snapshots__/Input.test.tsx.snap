// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "marginBottom": 20,
    }
  }
>
  <TextInput
    onChangeText={[Function]}
    placeholder="Email"
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderRadius": 8,
        "height": 40,
        "paddingLeft": 10,
        "width": 320,
      }
    }
    value="test@test.com"
  />
</View>
`;
